### ISSUE TITLE
Performance issues when retrieving subgroups - table (or index) full scan

### ISSUE THREAD

**Issue by rpetrescu** [keycloak/keycloak#21868]
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

storage

### Describe the bug

Named query "getGroupIdsByParent" from GroupEntity class triggers a table (or index) full scan on KEYCLOAK_GROUP table because the SIBLING_NAMES index (unique constraint) is not used. This is because the first element of the index (REALM_ID) is not sent as a parameter.

Depending the DB, a query will use (or not) an index if the leftmost column of the index is present in the query search criteria, that is why it is important to specify it when available.

Solution: add the realm Id as a parameter in the query in order to make of the index.

### Version

21.1.2

### Expected behavior

The SIBLING_NAMES index to be used in order to avoid a full table/index scan. 

### Actual behavior

The named query "getGroupIdsByParent" performs a full table/index scan on the KEYCLOAK_GROUP table. The performance impact is significant for 10K + groups in the database. The performance of the query is not influenced by the number of realms or how the groups are structured in subgroups.

### How to Reproduce?

Check the SQL execution plan for the following query (generated by the JPA layer):

```
SELECT
  `groupentit0_` . `ID` AS `col_0_0_`
FROM
  `KEYCLOAK_GROUP` `groupentit0_`
WHERE
  `groupentit0_` . `PARENT_GROUP` = ?
```

### Anything else?

The fix is very simple and does not have any side-effects.
I will prepare and submit a PR for this.

**Comment by ahus1** [comment]
Thank you for looking into this. A contribution for this is welcome. We're looking forward to your pull request.

**Comment by rpetrescu** [comment]
Anyway, here are the changes (in bold):

$ git show 4b957e51fba41c9ab995e0d558791f2f41b2429c
commit 4b957e51fba41c9ab995e0d558791f2f41b2429c
Author: Razvan Petrescu ***@***.***>
Date:   Fri Jul 21 12:47:41 2023 +0200

    Add the realm Id as a param to the named query getGoupIdsByParent in
order to use (or make better use of) the SIBLING_NAMES index on
KEYCLOAK_GROUP table.

    Closes #21868

diff --git
a/model/jpa/src/main/java/org/keycloak/models/jpa/GroupAdapter.java
b/model/jpa/src/main/java/org/keycloak/models/jpa/GroupAdapter.java
index 5fd5b23e41..b9471712ff 100755
--- a/model/jpa/src/main/java/org/keycloak/models/jpa/GroupAdapter.java
+++ b/model/jpa/src/main/java/org/keycloak/models/jpa/GroupAdapter.java
@@ -122,6 +122,7 @@ public class GroupAdapter implements GroupModel ,
JpaModel<GroupEntity> {
     @Override
     public Stream<GroupModel> getSubGroupsStream() {
         TypedQuery<String> query =
em.createNamedQuery("getGroupIdsByParent", String.class);
*+        query.setParameter("realm", group.getRealm());*
         query.setParameter("parent", group.getId());
         return
closing(query.getResultStream().map(realm::getGroupById).filter(Objects::nonNull));
     }
diff --git
a/model/jpa/src/main/java/org/keycloak/models/jpa/entities/GroupEntity.java
b/model/jpa/src/main/java/org/keycloak/models/jpa/entities/GroupEntity.java
index ba44986efa..6674d7b9cd 100755
---
a/model/jpa/src/main/java/org/keycloak/models/jpa/entities/GroupEntity.java
+++
b/model/jpa/src/main/java/org/keycloak/models/jpa/entities/GroupEntity.java
@@ -28,7 +28,7 @@ import java.util.LinkedList;
  * @version $Revision: 1 $
  */
 @NamedQueries({


*-        @NamedQuery(name="getGroupIdsByParent", query="select u.id
<http://u.id> from GroupEntity u where u.parentId = :parent"),+
 @NamedQuery(name="getGroupIdsByParent", query="select u.id <http://u.id>
from GroupEntity u where u.realm = :realm and u.parentId = :parent"),*
   @NamedQuery(name="getGroupIdsByRealm", query="select u.id from
GroupEntity u where u.realm = :realm  order by u.name ASC"),
         @NamedQuery(name="getGroupIdsByNameContaining", query="select u.id
from GroupEntity u where u.realm = :realm and u.name like
concat('%',:search,'%') order by u.name ASC"),
         @NamedQuery(name="getGroupIdsByNameContainingFromIdList",
query="select u.id from GroupEntity u where u.realm = :realm and lower(
u.name) like lower(concat('%',:search,'%')) and u.id in :ids order by u.name
ASC"),
